ChopChop/Storage/StorageManager.swift

```diff
@@ -228,6 +233,162 @@ extension StorageManager {
     }
 }
 
+// MARK: - Firebase operations
+extension StorageManager {
+
+    func publishRecipe(recipe: inout Recipe, userId: String) throws {
+        var cuisine = ""
+        if let categoryId = recipe.recipeCategoryId {
+            cuisine = (try? (fetchRecipeCategory(id: categoryId)?.name) ?? "") ?? ""
+        }
+        let steps = recipe.steps.map({ $0.content })
+        let ingredients = recipe.ingredients.map({
+            OnlineIngredientRecord(name: $0.name, quantity: $0.quantity.record)
+        })
+        let recipeRecord = OnlineRecipeRecord(
+            name: recipe.name,
+            creator: userId,
+            servings: recipe.servings,
+            cuisine: cuisine,
+            difficulty: recipe.difficulty,
+            ingredients: ingredients,
+            steps: steps
+        )
+        let onlineId = try firebase.addRecipe(recipe: recipeRecord)
+        recipe.onlineId = onlineId
+        try self.saveRecipe(&recipe)
+    }
+
+    // this should only be called once when the app first launched
+    func createUser(name: String) throws -> String {
```

Comment:
parameter should be the actual user object, so that the user object can do the checks (eg emptyName)